lib/src/repo.rs

```diff
@@ -732,11 +732,17 @@ pub struct MutableRepo {
     base_repo: Arc<ReadonlyRepo>,
     index: Box<dyn MutableIndex>,
     view: DirtyCell<View>,
-    parent_mapping: HashMap<CommitId, Vec<CommitId>>,
+    // TODO: make these fields private again
+    // The commit identified by the key has been replaced by all the ones in the value, typically
+    // because the key commit was abandoned (the value commits are then the abandoned commit's
+    // parents). A child of the key commit should be rebased onto all the value commits. A branch
+    // pointing to the key commit should become a conflict pointing to all the value commits.
+    pub(crate) parent_mapping: HashMap<CommitId, Vec<CommitId>>,
     /// Commits with a key in `parent_mapping` that have been divergently
     /// rewritten into all the commits indicated by the value.
-    divergent: HashSet<CommitId>,
-    abandoned: HashSet<CommitId>,
+    pub(crate) divergent: HashSet<CommitId>,
+    // Commits that were abandoned. Their children should be rebased onto their parents.
+    pub(crate) abandoned: HashSet<CommitId>,
```

Comment:
nit: if we keep `DescendantRebaser` and `rewrite` module as is, one option is to declare the data struct in `rewrite`, and let `MutableRepo` access it through pub methods.

Just an idea. Maybe there will be a better option (such as moving `rewrite` and `MutableRepo` under sub module.)

Reply:
I plan to move more logic into `MutableRepo`. Moving it into a new module probably makes sense because it's getting pretty large. I'll leave that until after all this refactoring is done if you don't mind.